Azure/azure-cli
# [Compute] disk create: Trusted launch; [Compute] disk create: Hibernation

**Description**

Resolve https://github.com/Azure/azure-cli/issues/17656
Resolve https://github.com/Azure/azure-cli/issues/17599

disk create, 2 new parameters, --security-type, --support-hibernation

Fix a field name error. It is caused by a SDK breaking change. `hyper_vgeneration` became `hyper_v_generation`

**Testing Guide**

```
        self.cmd('disk create -g {rg} -n d1 --image-reference Canonical:UbuntuServer:18.04-LTS:18.04.202002180 --security-type TrustedLaunch', checks=[
            self.check('securityProfile.securityType', 'TrustedLaunch')
        ])
        self.cmd('disk create -g {rg} -n d1 --size-gb 10 --support-hibernation true', checks=[
            self.check('supportsHibernation', True)
        ])
```

**History Notes**


[Compute] disk create: Trusted launch
[Compute] disk create: Hibernation

---

This checklist is used to make sure that common guidelines for a pull request are followed.
